src/app/products/product-filter/product-filter.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import {
+  map,
+  startWith,
+  debounceTime,
+  distinctUntilChanged,
+} from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-product-filter',
+  templateUrl: './product-filter.component.html',
+  styleUrls: ['./product-filter.component.scss'],
+})
+export class ProductFilterComponent implements OnInit {
+  private queryFieldSubscription: Subscription;
+  @Output()
+  searchValue = new EventEmitter<String>();
+  queryField = new FormControl();
+
+  constructor() {}
+
+  ngOnInit(): void {
+    this.queryFieldSubscription = this.queryField.valueChanges
+      .pipe(
+        startWith(''),
+        debounceTime(400),
+        distinctUntilChanged(),
+        map((value) => this._filter(value))
+      )
+      .subscribe();
```

Comment:
czemu search nie wykonuje sie w subscribe? map powinien sluzyc to mapowania a nie do wykonywania operacji.

Reply:
poprawione